daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -87,6 +87,7 @@ class Engine(val config: EngineConfig = EngineConfig.Stable) {
     * The resulting transaction is annotated with packages required to validate it.
     */
   def submit(
+      submitters: Set[Party],
```

Comment:
Note that this changes the API for many ledger integrations. 
@remyhaemmerle-da: would you say we should keep the submitters in the LF `Commands` data type? They basically represent the parties for the root node authorization, right?

We could look at it like this:
lf.Commands: root node authorization (aka `actAs` parties)
domain.Commands: `actAs ∪ readAs` for contract lookup

Reply:
@andreaslochbihler-da @simonmaxen-da  is this change annoying to you ?

Reply:
For reference, the proposed change is to remove the submitting parties from `com.daml.lf.command.Commands`, and instead pass them as a separate parameter to `Engine.submit`.

Reply:
The motivation for keeping `actAs` and `readAs` sets of parties on the API level was that the `Engine` should not be aware of such concepts. Is such assumption valid?

Reply:
We don't need `submit` in Canton. Submission is taken care of by the Ledger API server at the moment. So no problem from the Canton side.

Reply:
I was more worried in the change in`com.daml.lf.command.Commands`. Since @kamil-da already change the signature of `com.daml.lf.command.Commands` in #8081 I do not think it should be a heartbreaking problem.

Reply:
That should be fine as well. (I've already adapted to #8081 in Canton.)